bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/discovery/HomeAssistantDiscovery.java

```diff
@@ -166,8 +168,21 @@ public void receivedMessage(ThingUID connectionBridge, MqttBrokerConnection conn
             thingIDPerTopic.put(topic, thingUID);
 
             // We need to keep track of already found component topics for a specific thing
-            Set<HaID> components = componentsPerThingID.computeIfAbsent(thingID, key -> ConcurrentHashMap.newKeySet());
-            components.add(haID);
+            final List<HaID> components;
+            {
```

Comment:
Why a new block here?

Reply:
To prevent accidental usage of the temporary local variable `unsortedComponents`

Reply:
Ok even if it looks to me unusual,